src/theory/sets/cardinality_extension.h

```diff
@@ -324,18 +324,22 @@ class CardinalityExtension
   void checkNormalForm(Node eqc, std::vector<Node>& intro_sets);
 
   /**
-   * add cardinality lemmas for each finite type
+   * add cardinality lemmas for each type except uninterpreted sorts with
+   * finite-model-find
    */
-  void checkFiniteTypes();
+  void checkUnivsetCardinality();
```

Comment:
We should have a better name for this function.  Maybe `checkCardinalityExtended`  for checking cardinality extended (handling cases that go beyond Bansal et al 2016).

Reply:
Sure